app/services/s3_zip_downloader.rb

```diff
@@ -14,10 +14,11 @@ def generate!
   private
 
   def build_zip_file(documents, bundle)
+    filename_counts = {}
     Dir.mktmpdir("#{@claim.case_number}-") do |tmp_dir|
       Zip::File.open(bundle, Zip::File::CREATE) do |zip_file|
         documents.map(&:document).each do |document|
-          zip_file.add(document.filename, local_file(document, tmp_dir))
+          zip_file.add(get_filename(document.filename, filename_counts), local_file(document, tmp_dir))
         end
```

Comment:
Did the previous incarnation of this not work. i prefer its simplicity


```
documents.map(&:document).each_with_index do |document, i|
  zip_file.add("#{i}_#{document.filename}", local_file(document, tmp_dir))
end
```
or similar that you had

Reply:
It did work but I thought that the `filename (1).doc` is closer to what people expect when downloading a second file with the same name. Adding the index to every file implies an ordering (although this probably isn't really a problem).

Reply:
I have changed it back now.